flink-runtime/src/main/java/org/apache/flink/runtime/state/StateObject.java

```diff
@@ -63,4 +63,14 @@
      * @return Size of the state in bytes.
      */
     long getStateSize();
+
+    /**
+     * Returns the incremental state size in bytes. If the size is unknown, this method would return
+     * same result as {@link #getStateSize()}.
+     *
+     * @return Size of incremental state in bytes.
+     */
+    default long getIncrementalStateSize() {
+        return getStateSize();
+    }
```

Comment:
Even applying to `CompositeStateHandle`, I think we still need to add a default method of `getIncrementalStateSize`.

Reply:
Do we? Default implementation allows to forget to implement it, and that's probably the case with `SubtaskState`, isn't it?

The interface `CompositeStateHandle` is a bit broken I think because it combines:
a) marking composites and
b) registering shared states
Which are not directly related (so it should probably be split into two)

So less invasive and safe change is to move the method to `CompositeStateHandle` but remove `default`.

Reply:
I have tried to refactor to add a new non-default method in `CompositeStateHandle`.

Reply:
Thanks!

Reply:
Thanks!

Reply:
Thanks!

Reply:
Thanks!